editor/editor/layout.js

```diff
@@ -17,8 +17,7 @@ class Layout extends wp.element.Component {
 	}
 
 	switchMode( newMode ) {
-		// TODO: we need a serializer from blocks here
-		const html = this.state.html;
+		const html = this.mode === 'text' ? this.state.html : wp.blocks.serialize( this.state.blocks );
```

Comment:
I'm performing this logic here for performance reasons. Only parse/serialize when switching.

And good catch for the `this.mode` error